lib/src/async_expand.dart

```diff
@@ -32,9 +32,8 @@ extension AsyncExpand<T> on Stream<T> {
   /// -  If [convert] returns a single subscription stream it may be listened to
   /// and never canceled.
   /// -  For any period of time where there are no listeners on the result
-  /// stream, any sub streams from previously emitted events will be ignored,
-  /// regardless of whether they emit further events after a listener is added
-  /// back.
+  /// stream, any event on the source stream is ignored and will not be passed
+  /// to [convert].
```

Comment:
> I wouldn't delay subscribing to non-broadcast streams.

Why is that?

Reply:
If the returned stream is a broadcast stream, then it shouldn't matter if you cancel and re-listen, or stay subscribed and just throw away events for the same duration, it should give the same result (aka: the stream creates the same events at the same time, independently of whether there are listeners, it's purely an optimization to not call `add` while there is no listener.)
In that case, it's OK to not subscribe to inner broadcast streams until you need them, because it won't matter for which event subscribers see. You're using the broadcast property to unsubscribe until you need events again, instead of discarding them. For a properly implemented broadcast stream, that should be safe.

It does matter when you subscribe to an inner non-broadcast stream, because it changes the timing of events, so you should subscribe to it at the same time, aka. immediately, whether you have listeners or not. If you don't, the returned stream does not behave properly as a broadcast stream. It may not be obvious to listeners, because they don't know which events they should get, but we know! :)

Reply:
I agree, it's better that time of presence/absence of _other listeners_ on the stream do not impact the events that are seen. I updated the behavior so immediately listen on broadcast streams, and to avoid ever canceling the outer subscription. I updated the doc comment to warn about this behavior and added new tests.